src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs

```diff
@@ -185,19 +183,32 @@ async Task PresentModal(Page modal, bool animated)
 
 			if (animated)
 			{
-				dialogFragment!.AnimationEnded += OnAnimationEnded;
+				TaskCompletionSource<bool> animationCompletionSource = new();
+				
+				dialogFragment.AnimationEnded += OnAnimationEnded;
+
+				void OnAnimationEnded(object? sender, EventArgs e)
+				{
+					dialogFragment.AnimationEnded -= OnAnimationEnded;
+					animationCompletionSource.SetResult(true);
+				}
 
 				await animationCompletionSource.Task;
 			}
 			else
 			{
-				animationCompletionSource.TrySetResult(true);
-			}
+				// Non-animated modals need to wait for presentation completion to prevent race conditions
+				TaskCompletionSource<bool> presentationCompletionSource = new();
```

Comment:
This could be only a `TaskCompletionSource`, since the value isn't needed

Reply:
I have modified the code based on your suggestion. Changed to TaskCompletionSource without generic type parameter in the non-animated modal presentation path as the completion value isn't needed.

Reply:
@pictos, Thanks for clarifying. I have maintained the `TaskCompletionSource<bool>` with the generic type parameter since this project targets .NET Standard 2.0.